specs/interop/token-bridging.md

```diff
@@ -72,26 +101,200 @@ contract OptimismSuperchainERC20 is ERC20, ISuperchainERC20 {
+
+```solidity
+function xApprove(address _spender, uint256 _chainId, uint256 _amount) external returns (bool) {
+  // TODO [research] _chainId wildcard as type(uint256).max (this is very risky)
+  _xAllowances[msg.sender][_spender][_chainId] = _amount;
+  emit XApproval(msg.sender, _spender, _chainId, _amount);
+  return true;
+}
+// this will override the ERC20 view function
+function allowance(address _owner, address _spender) external view returns (uint256) {
+  return allowance(_owner, _spender, block.chainid);
+}
+
+function allowance(address _owner, address _spender, uint256 _chainId) external view returns (uint256) {
+  if (_chainId == block.chainId) return _allowance[_owner][_spender];
+  return _xAllowance[_owner][_spender][_chainId];
+}
+```
+
+It is possible for the standard to also have remote approvals. This feature, however, was discarded as an address might have different owners in different chains (think of smart wallets for instance). Moreover, even if this was not an issue, switching `chainId` is not that relevant UX-wise to include a dedicated function in the token standard.
+
+**`remoteTransferFrom()`**
+
+As already mentioned, the `remoteTransferFrom()` \*\*\*\*will allow using funds on a remote chain on behalf of another party. The function should initiate a call to the `L2ToL2CrossChainMessenger` without burning anything on the origin chain.
+
+To process a `remoteTransferFrom()` on the destination, the standard should include a `relayTransferFrom()` function that checks local allowance and calls `burn()`. Then, it will check if the target lives on the same `chainId` to call `relayAndExecute()` to the corresponding contract.
+
+Finally, the standard should included a `relayAndExecute()` function that might be used to resume execution on the origin chain. This is how the atomicity lost in the `transferFrom()` gets addressed.
+
+A possible implementation for these functionalities would look like this:
+
+```solidity
+function remoteTransferFrom(
+  address _owner,
+  address _recipient,
+  uint256 _spendChainId,
+  uint256 _executeChainId,
+  uint256 _amount,
+  bytes memory _data
+) external returns (bool) {
+  bytes memory _message =
+    abi.encodeCall(this.relayTransferFrom, (_owner, msg.sender, _recipient, block.chainid, _executionChainId, _amount, _data));
+  _sendMessage(_spendChainId, _message);
+  return true;
+}
+
+function relayTransferFrom(
+  address _owner,
+  address _spender,
+  address _recipient,
+  uint256 _originChainId,
+  uint256 _executionChainId,
+  uint256 _amount,
+  bytes memory _data
+) external onlyMessenger {
+  uint256 allowed = _xAllowances[_owner][_spender][_originChainId];
+
+  if (allowed != type(uint256).max) {
+    _xAllowances[_owner][_spender][_originChainId] = allowed - _amount;
+  }
+
+  _burn(_owner, _amount);
+
+  if (block.chain == _executionChainId){
+    _relayAndExecute(_owner, _recipient, _spender, _originChainId, block.chainid, _amount, _data);
+  } else {
```

Comment:
I am confused by this if else branch, can you explain it?

Reply:
The recipient might be in a chain which is different from where the funds are (where the `relayTransferFrom` call is happening). This if will check if its on the same chain (and perform an internal call) or not (and craft a message to execute the call cross-chain).

Reply:
to clarify, the first if corresponds to:
Contract in chain A calls remoteTransferFrom to chain B with a recipient in chain B 
The second corresponds to:
Contract in chain A calls remoteTransferFrom to chain B with a recipient in chain C

Reply:
closing for now, as we are not including this flow for the moment.